src/components/spreadsheet/spreadsheet.ts

```diff
@@ -314,6 +317,12 @@ export class Spreadsheet extends Component<SpreadsheetProps, SpreadsheetChildEnv
         this.composerFocusStore.focusGridComposerCell(content),
     });
 
+    this.notificationStore.updateNotificationCallbacks({ ...this.props });
+
+    onWillUpdateProps((nextProps) => {
+      this.notificationStore.updateNotificationCallbacks({ ...nextProps });
```

Comment:
I think this is called at every rendering, whether the props actually changed or not.
Should we check that they changed `this.props !== nextProps` ? (not sure the reference are still the same if nothing changed)